### PR TITLE
Trash functionality and note color

### PR DESCRIPTION
**Pull request by dusariudaykumar**
# Feat Trash Functionality And Color to Note
## Added
- Move to trash
- Delete from trash
- Color to note

### CODE REVIEW

File: src/Components/InputField/NotesField.jsx [hunk 1]
```diff
@@ -59,11 +80,23 @@ const NotesField = ({ notes, setNotes, addNoteHandler, updatehandler }) => {
...
-              <div className="notes-action-btn-wrapper">
-                <span className="material-icons">palette</span>
+              <div className="notes-action-btn-wrapper flex">
+                <span
+                  className="material-icons"
+                  onClick={() => colorPalletHandler()}>
+                  palette
+                </span>
+                {isColorPalletVisible && (
+                  <div>
```
Comment (devdeadviz) [inline src/Components/InputField/NotesField.jsx]: You can avoid use of div here as it's not required.

File: src/Pages/HomePage/Homepage.jsx [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import React, { useState } from "react";
+import { toast } from "react-toastify";
 import { NotesCard, NotesField } from "../../Components";
 import { useAuth } from "../../contexts";
 import { useNotes } from "../../contexts/notes-context";
```
Comment (devdeadviz) [inline src/Pages/HomePage/Homepage.jsx]: You can import all the contexts from the same file. Check it once!

File: src/Pages/TrashPage/TrashPage.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    noteDispatch,
+  } = useNotes();
+  const deleteFromTrashhandler = (notes) => {
+    const data = trash.filter((note) => note._id !== notes._id);
+    noteDispatch({ type: "DELETE_FROM_TRASH", payload: data });
+    toast.success("Deleted Note From Trash ");
+  };
+  const restoreFromTrash = async (notes) => {
+    const response = await addNotesService(notes, encodedToken);
+    console.log(response);
```
Comment (devdeadviz) [inline src/Pages/TrashPage/TrashPage.jsx]: Remove console statements.

File: src/reducers/noteReducer.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 export const noteReducer = (state, action) => {
-  console.log(state, action);
+  console.log(action);
```
Comment (devdeadviz) [inline src/reducers/noteReducer.js]: Same here

File: src/reducers/noteReducer.js [hunk 5]
```diff
@@ -41,6 +45,22 @@ export const noteReducer = (state, action) => {
...
+    case "COLOR_PALLET_VISIBLE":
+      return {
+        ...state,
+        isColorPalletVisible: !state.isColorPalletVisible,
+      };
+    case "PICK_COLOR":
+      return {
+        ...state,
+        isNoteColor: !state.isNoteColor,
+        // isColorPalletVisible: !state.isColorPalletVisible,
```
Comment (devdeadviz) [inline src/reducers/noteReducer.js]: Remove commented code.

**Review by devdeadviz — APPROVED** [review]
The code is clean and the file structure is also good. Gook work!
Great use of barrel exports.

File: src/Components/InputField/NotesField.jsx [hunk 6]
```diff
@@ -20,6 +39,7 @@ const NotesField = ({ notes, setNotes, addNoteHandler, updatehandler }) => {
   return (
     <>
       <form
+        style={{ background: noteBgColor }}
         className="notes-field-wrapper "
         onSubmit={
           !editNote
```
Comment (Chiragmodi01) [inline src/Components/InputField/NotesField.jsx:46]: onSubmit function seems to be missing

File: src/Components/InputField/NotesField.jsx [hunk 7]
```diff
@@ -30,11 +50,12 @@ const NotesField = ({ notes, setNotes, addNoteHandler, updatehandler }) => {
...
+                autocomplete="off"
                 type="text"
                 name="notesTitle"
                 className="notes-field-title"
                 placeholder="Title"
-                value={notes.notesTitle}
+                value={notesTitle}
                 onChange={changeHandler}
               />
               <span className="material-icons-outlined">push_pin</span>
```
Comment (Chiragmodi01) [inline src/Components/InputField/NotesField.jsx:63]: Pin feature doesn't seem to work

File: src/Components/NotesCard/NotesCard.css [hunk 8]
```diff
@@ -18,10 +19,14 @@
 .card-description {
   letter-spacing: 1px;
   font-weight: 500;
+  overflow-wrap: break-word;
+  word-wrap: break-word;
+  white-space: pre-wrap;
 }
 .card-btns {
   justify-content: end;
```
Comment (Chiragmodi01) [inline src/Components/NotesCard/NotesCard.css]: *flex-end

File: src/Components/NotesCard/NotesCard.jsx [hunk 9]
```diff
@@ -23,38 +28,54 @@ const NotesCard = ({
...
-                onClick={() => addArchiveHandler(notes)}>
-                archive
-              </span>
+                <span className="material-icons">label</span>
+                {path !== "/archive" ? (
+                  <span
+                    className="material-icons"
+                    onClick={() => addArchiveHandler(notes)}>
+                    archive
+                  </span>
```
Comment (Chiragmodi01) [inline src/Components/NotesCard/NotesCard.jsx]: You could just create separate cards for trash and archive, the code will looks more cleaner. (your preference!)

**Review by Chiragmodi01 — COMMENTED** [review]
- Not able to close the create-note-form after i open it.
- Note card seems to have extra height than expected, icons should be a the bottom of the note.
- User should not be able to create empty note.
- Looks like your main container of trash are not flex box. That's why cards are behaving as block elements, leaving space on sides.
Overall UI looks good.

File: src/Components/InputField/NotesField.jsx (hunk 7 above)
Comment (dusariudaykumar) [reply src/Components/InputField/NotesField.jsx:63]: i haven't added the pin functionality

File: src/Components/InputField/NotesField.jsx (hunk 6 above)
Comment (dusariudaykumar) [reply src/Components/InputField/NotesField.jsx:46]: I have added onSubmit functionality can you check correctly.
so, based on the ```editNote``` I'm adding a function

**Pull request merged**